### PR TITLE
add support for dns zones

### LINKED ISSUE #182: Support cloud dns
```
module "dns" {
  source  = "terraform-google-modules/cloud-dns/google"
  version = "3.0.1"

  name       = var.dns_name
  project_id = var.project_id
  type       = "public"
  domain     = var.dns_domain

  recordsets = [{
    name = "foo"
    type = "A"
    ttl  = 30
    records = [
      google_compute_global_address.foo.address,
    ]
  }]
}```

### PR DESCRIPTION
**Pull request by umairidris**
Fix #182

### CODE REVIEW

File: examples/tfengine/team.hcl [hunk 1]
```diff
@@ -378,6 +378,19 @@ template "project_apps" {
         iam_members = {
           "roles/container.viewer" = ["group:example-viewers@example.com"]
         }
+        dns_zones = [{
```
Comment (MartinPetkov) [inline examples/tfengine/team.hcl:381]: Why "zones"?

Comment (MartinPetkov) [reply examples/tfengine/team.hcl:381]: Looked around the module. Seems like that's just the terminology.

File: templates/tfengine/recipes/resources.hcl [hunk 2]
```diff
@@ -434,6 +448,74 @@ schema = {
...
+          }
+          resource_name = {
+            description = <<EOF
+              Override for Terraform resource name. If unset, defaults to normalized name.
+              Normalization will make all characters alphanumeric with underscores.
+            EOF
+            type        = "string"
+          }
+          domain = {
+            description = "Domain of DNS zone. Must end with period."
```
Comment (MartinPetkov) [inline templates/tfengine/recipes/resources.hcl:474]: You should add a regex to verify that it ends with a period.

File: templates/tfengine/recipes/resources.hcl [hunk 3]
```diff
@@ -904,6 +986,12 @@ template "compute_routers" {
 }
 {{end}}
 
+{{if has . "dns_zones"}}
+template "compute_routers" {
```
Comment (MartinPetkov) [inline templates/tfengine/recipes/resources.hcl:990]: Typo, should be "dns_zones"

File: templates/tfengine/recipes/resources.hcl [hunk 4]
```diff
@@ -434,6 +448,74 @@ schema = {
...
+    dns_zones = {
+      description = "https://github.com/terraform-google-modules/terraform-google-cloud-dns"
+      type        = "array"
+      items = {
+        type = "object"
+        additionalProperties = false
+        required = [
+          "name",
+          "domain",
+          "type",
```
Comment (xingao267) [inline templates/tfengine/recipes/resources.hcl:460]: add record set as well as it's using `hcl` (mandatory) above.

File: templates/tfengine/recipes/resources.hcl (hunk 2 above)
Comment (umairidris) [reply templates/tfengine/recipes/resources.hcl:475]: Done.

File: templates/tfengine/recipes/resources.hcl (hunk 3 above)
Comment (umairidris) [reply templates/tfengine/recipes/resources.hcl]: Done.

File: templates/tfengine/recipes/resources.hcl (hunk 4 above)
Comment (umairidris) [reply templates/tfengine/recipes/resources.hcl:460]: Done.

**Pull request merged**